mj_envs/envs/arms/push_base_v0.py

```diff
@@ -16,32 +18,68 @@ class PushBaseV0(env_base.MujocoEnv):
     }
 
 
-    def __init__(self, model_path, config_path, robot_site_name, object_site_name,
-                target_site_name, target_xyz_range, **kwargs):
-        # get sims
-        self.sim = env_base.get_sim(model_path=model_path)
-        self.sim_obsd = env_base.get_sim(model_path=model_path)
+    def __init__(self, 
+                model_path, 
+                config_path, 
+                robot_site_name,
+                object_site_name,
+                target_site_name, 
+                target_xyz_range,
+                seed = None,
+                **kwargs):
+
+        # EzPickle.__init__(**locals()) is capturing the input dictionary of the init method of this class.
+        # In order to successfully capture all arguments we need to call gym.utils.EzPickle.__init__(**locals())
+        # at the leaf level, when we do inheritance like we do here.
+        # kwargs is needed at the top level to account for injection of __class__ keyword.
+        # Also see: https://github.com/openai/gym/pull/1497
+        gym.utils.EzPickle.__init__(**locals())
+
+        # This two step construction is required for pickling to work correctly. All arguments to all __init__ 
+        # calls must be pickle friendly. Things like sim / sim_obsd are NOT pickle friendly. Therefore we 
+        # first construct the inheritance chain, which is just __init__ calls all the way down, with env_base
+        # creating the sim / sim_obsd instances. Next we run through "setup"  which relies on sim / sim_obsd
+        # created in __init__ to complete the setup.
+        super().__init__(model_path=model_path)
```

Comment:
Kwargs should be passes here. Otherwise we might not be able to pass any arguments to subcomponents like self.robot
https://github.com/vikashplus/mj_envs/blob/dambrl/mj_envs/envs/env_base.py#L76

Reply:
Let us discuss offline. Passing kwargs to init will cause pickle issues.